libdrgn/program.c

```diff
@@ -584,6 +577,37 @@ drgn_program_set_core_dump(struct drgn_program *prog, const char *path)
 	return err;
 }
 
+LIBDRGN_PUBLIC struct drgn_error *
+drgn_program_set_fd(struct drgn_program *prog, int fd)
+{
+	struct drgn_error *err;
+	char path[sizeof("/proc/self/fd/") + 20];
+
+	err = drgn_program_check_initialized(prog);
+	if (err)
+		return err;
+
+	snprintf(path, sizeof(path), "/proc/self/fd/%d", fd);
```

Comment:
This path is only used for error reporting, right?

This is how we do this statically-sized buffer for `printf()` in other places:

```suggestion
	err = drgn_program_check_initialized(prog);
	if (err)
		return err;

	#define FORMAT "/proc/self/fd/%d"
	char path[sizeof(FORMAT) - sizeof("%d") + max_decimal_length(int) + 1];
	snprintf(path, sizeof(path), FORMAT, fd);
	#undef FORMAT
```

Reply:
Nice, I just manually did the max decimal length of uint64_t and added one. Should have known you had a macro for it... :P